### DISCUSSION TITLE
GAT(K) command doesn't increment get_hits

### DISCUSSION THREAD

**Discussion by tomko72** [memcached/memcached#1124] [Q&A]
Hi, we are using memcached quite extensively, but have found out just recently that using get and touch (GAT) command doesn't increment `get_hits` stats. Is this by design/convention?
As an outsider I'd expect both counters to increment (`touch_hits` and `get_hits`) in that case.
I've traced the issue to the source code https://github.com/memcached/memcached/blob/77709d04dd4fc7d59f59425802cb9645b2cfe0f8/proto_bin.c#L497
and propose changing `else` condition to `if (should_return_value)`. 
I understand there may be more considerations in place, it just struck me as an unintuitive approach.
Thanks, Tom

**Comment by dormando** [comment]
Hi,

I'll have to take a more careful look. That code's been that way forever and I don't know the rationale offhand. It's also difficult to change pre-existing counter behavior because it can set off people's alarms when numbers suddenly change.

On the surface it does look like the GAT commands just emulate touch counters, I just don't know why or if that's correct or if it's the same in all of the protocols offhand.

**Reply by tomko72** [reply]
I understand it needs careful approach. Thanks for your consideration anyway. 
Tom
